### PR TITLE
Jupyter add svc pipelines

### PR DESCRIPTION
**Pull request by chainlink**
Adds the ability to create service pipelines in the Jupyter extension
[INT-1043]

[INT-1043]: https://pachyderm.atlassian.net/browse/INT-1043?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jupyter-extension/jupyterlab_pachyderm/pps_client.py [hunk 1]
```diff
@@ -205,17 +209,26 @@ def create_pipeline_spec(config: PpsConfig, companion_branch: str) -> dict:
     pipeline = dict(name=config.pipeline.name)
     if config.pipeline.project and config.pipeline.project.name:
         pipeline['project'] = dict(name=config.pipeline.project.name)
-    return dict(
+    piplineSpec =  dict(
```
Comment (smalyala) [inline jupyter-extension/jupyterlab_pachyderm/pps_client.py:212]: "piplineSpec" -> "pipelineSpec" here and below :)

File: jupyter-extension/src/plugins/mount/components/Pipeline/Pipeline.tsx [hunk 2]
```diff
@@ -157,6 +159,21 @@ const Pipeline: React.FC<PipelineProps> = ({
           placeholder={placeholderRequirements}
         ></input>
       </div>
+      <div className="pachyderm-pipeline-input-wrapper">
+        <label className="pachyderm-pipeline-input-label" htmlFor="port">
+          *Port:{'  '}
```
Comment (smalyala) [inline jupyter-extension/src/plugins/mount/components/Pipeline/Pipeline.tsx:164]: Port shouldn't be a required field right?

File: jupyter-extension/jupyterlab_pachyderm/pps_client.py [hunk 3]
```diff
@@ -205,17 +209,26 @@ def create_pipeline_spec(config: PpsConfig, companion_branch: str) -> dict:
...
         ),
+
         input=input_spec,
         update=True,
         reprocess=True,
     )
+    if config.port:
+        piplineSpec['service'] = dict(
+            external_port=config.port,
+            internal_port=config.port,
```
[the remark this answers is not in the document]
Comment (chainlink) [reply jupyter-extension/jupyterlab_pachyderm/pps_client.py:227]: Potentially! But I think we could add it as a follow on task if requested.

**Review by smalyala — APPROVED** [review]
LGTM

**Review by bbonenfant — APPROVED** [review]

**Pull request merged**